source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object_collection/rigid_object_collection.py

```diff
@@ -0,0 +1,480 @@
+            if len(matching_prims) == 0:
+                raise RuntimeError(f"Could not find prim with path {rigid_object_cfg.prim_path}.")
+
+        # stores object names
+        self._object_names_list = []
+
+        # note: Use weakref on all callbacks to ensure that this object can be deleted when its destructor is called.
+        # add callbacks for stage play/stop
+        # The order is set to 10 which is arbitrary but should be lower priority than the default order of 0
+        timeline_event_stream = omni.timeline.get_timeline_interface().get_timeline_event_stream()
+        self._initialize_handle = timeline_event_stream.create_subscription_to_pop_by_type(
+            int(omni.timeline.TimelineEventType.PLAY),
+            lambda event, obj=weakref.proxy(self): obj._initialize_callback(event),
+            order=10,
+        )
+        self._invalidate_initialize_handle = timeline_event_stream.create_subscription_to_pop_by_type(
+            int(omni.timeline.TimelineEventType.STOP),
+            lambda event, obj=weakref.proxy(self): obj._invalidate_initialize_callback(event),
+            order=10,
+        )
+
+        self._debug_vis_handle = None
+
+    """
+    Properties
+    """
+
+    @property
+    def data(self) -> RigidObjectCollectionData:
+        return self._data
+
+    @property
+    def num_instances(self) -> int:
+        return self.root_physx_view.count // self.num_objects
+
+    @property
+    def num_objects(self) -> int:
+        """Number of objects in the asset.
+
+        This corresponds to the distinct number of rigid bodies in the asset.
+        """
+        return len(self.object_names)
+
+    @property
+    def object_names(self) -> list[str]:
+        """Ordered names of bodies in the rigid object collection."""
+        return self._object_names_list
+
+    @property
+    def root_physx_view(self) -> physx.RigidBodyView:
+        """Rigid body view for the rigid body collection (PhysX).
+
+        Note:
+            Use this view with caution. It requires handling of tensors in a specific way.
+        """
+        return self._root_physx_view
+
+    """
+    Operations.
+    """
+
+    def reset(self, env_ids: torch.Tensor | None = None, object_ids: slice | torch.Tensor | None = None):
+        """Resets all internal buffers of selected environments and objects.
+
+        Args:
+            env_ids: The indices of the object to reset. Defaults to None (all instances).
+            object_ids: The indices of the object to reset. Defaults to None (all objects).
+        """
+        # resolve all indices
+        if env_ids is None:
+            env_ids = self._ALL_ENV_INDICES
+        if object_ids is None:
+            object_ids = self._ALL_OBJ_INDICES
+        # reset external wrench
+        self._external_force_b[env_ids[:, None], object_ids] = 0.0
+        self._external_torque_b[env_ids[:, None], object_ids] = 0.0
```

Comment:
this code will break if we don't have the exact same number of rigid objects as the defined ``num_envs`` of our interactive scene. Would recommend that we remove them

Reply:
Where should we put them though?

Reply:
I am not sure if we can. As env_ids assumes that we have num_envs instances but that may not be the case, we might have to leave that reset to the user?

Reply:
This argument also holds for Articulations and RigidObjects thought

Reply:
true that, but wasn't the idea that with the collection, we allow exactly an option that does not have to be repeated num_envs times?